Feat/new-user-stories

ese-varo opened the pull request:
## User stories for this feature
- #### As a registered user, I want to see a list of all the registered users and if I already have a conversation with them
  - See a list of all the users
  - Have a numeric indicator of how many conversations have with that user
    - You can display the conversations and show the name of the same that redirects you to the chat
  - If the registered user does not have any conversation with the user, you can click to create the conversation and redirects to it

- #### As a registered user, when it creates a conversation it has to add the description
  - At the moment when the user creates the conversation, he/she has to register emoji, title and description
    - Title, emoji, and description are required
    - Error handling
    - If the emoji is not found or represents an error, save an emoji by default
  - All the conversations have a profile
    - The profile consists of a title, emoji, and description
    - If the conversations are private, only the participants can see the profile
    - If the conversations are public, everyone can see the profile

- #### As an unregistered user, I can see a list of the last 5 conversations to know if I am interested in registering
  - In the login or a separated page, I can see a list of the last 5 most active public conversations (that have a greater number of messages and are not more than a month ago)
  - He/she can only see the profile of the conversation

ese-varo committed:
Add query object for public conversations

ese-varo committed:
Implement form

ese-varo committed:
Add meet others functionality

ese-varo committed:
Add services

ese-varo committed:
Create partial for public conversations

ese-varo committed:
Add conversation query

ese-varo committed:
Refactor alert messages

ese-varo committed:
Add error handling functionality

ese-varo committed:
Refactor conversations query

jvargas98 commented on app/controllers/users_controller.rb:
```diff
@@ -1,10 +1,7 @@
-
   def new
     @user = User.new
+    # get '*path', to: redirect('/')
```
maybe you can remove this comment if you don't need it anymore

jvargas98 commented on app/javascript/packs/conversation_show.js:
```diff
@@ -0,0 +1,31 @@
+window.addEventListener("load", () => {
+  const queryString = window.location
+  console.log(queryString.href)
+  // setInterval(get_messages(queryString.href), 3000); 
+  // setInterval(() => { get_messages(queryString.href); }, 3000);
```
The same here

jvargas98 commented on app/javascript/packs/conversation_show.js:
```diff
@@ -0,0 +1,31 @@
+  console.log(data)
+}
+
+// async function getalarms() {
+//   const url = BASE_URL + "alarms.json";
+//   const response = await fetch(url, options);
+//   const data = await response.json();
+//   evaluateAlarms(data);
+// }
```
Same

jvargas98 commented on app/models/message_observer.rb:
```diff
@@ -2,7 +2,7 @@ class MessageObserver < ActiveRecord::Observer
     message.conversation.users.each do |user|
       next if user == message.user
-      ConversationMailer.new_message_email(user, message.conversation).deliver_now
+      # ConversationMailer.new_message_email(user, message.conversation).deliver_now
```
maybe you can remove this comment, but I think you need it.

jvargas98 commented on app/views/conversations/profile.html.erb:
```diff
@@ -0,0 +1,25 @@
+          <%= presenter.emoji %> 
+        </h1>
+        <div class="col">
+          <% if current_user %>
+            <%= link_to 'Back', @conversation, class: 'btn btn-secondary float-right' %>
+          <% else %>
+            <%= link_to 'Back', root_path, class: 'btn btn-secondary float-right' %>
+          <% end %>
```
I'm not sure but maybe a ternary operator will work, I think this because only change the path

GregVaz replied:
It's correct, you can change the path with a ternary operator

ese-varo replied:
I'll try that, thanks!

jvargas98 reviewed (commented):
Great job, I liked how you add the services and object queries. I left you some comments and I also wish you could implement partial views to clean up your views a little more.

ese-varo committed:
Translate layout erb files into haml

GregVaz commented on app/views/layouts/_alerts.html.erb:
```diff
@@ -0,0 +1,7 @@
+  <% else %>
+      <%= content_tag :div, msg, class: "alert alert-warning" %>
+  <% end %>
+<% end %>
```
Just an idea
If you want to reduce this code, you could put the name of the _type of alert_ as the suffix of alert, you could establish your own css class and that's all
`<%= content_tag :div, msg, class: "alert alert-#{name}"`

ese-varo replied:
I actually tried that way, but it seems that rails doesn't let you use create custom alerts as `danger, primary, success, info` things like that which are the classes that bootstrap provides you to add styles to alerts. That's why I'm using the else option as a fallback for other than `success` flash messages. 
In order to implement the approach that you mention I should create custom classes that could match with the rails flash options, may be I could give it a try :sweat_smile:

LoginALee commented:
Good work, i liked how you implemented error handling :+1:

ese-varo committed:
Translate conversation erb files into haml

ese-varo committed:
Refactor conversation presenter (dry)

ese-varo committed:
Fix messages emoji display issue

The messages didn't show the emojis right after created them because an
after_action callback was implemented in the messages controller, it was
fixed calling the FillerEmoji service right after creating the message
and before responding to requests.

ese-varo committed:
Refactor conversations query

ese-varo committed:
Implement haml on views

ese-varo committed:
Add pr requested changes

ese-varo committed:
Refactor meet others logic

jvargas98 reviewed (approved):
LGTM 🚀
